landlab/components/stream_power/cfuncs.pyx

```diff
@@ -4,6 +4,9 @@ cimport cython
 
 #from libc.math cimport fabs
 
+# as best as KRB can tell, the function _brentq is in c.
+# not sure what, if anything I need to do to 
+from scipy.optimize._zeros import _brentq as brentq
 
```

Comment:
I suspect the way you've imported `_brentq` isn't quite right and is slowing things down. Perhaps try:
```python
from scipy.optimize._zeros cimport _brentq as brentq
```
and see if that speeds things up at all.